x/mongo/driver/operation.go

```diff
@@ -830,7 +812,7 @@ func (op Operation) retryable(desc description.Server) bool {
 			return true
 		}
 		if retryWritesSupported(desc) &&
-			desc.WireVersion != nil && desc.WireVersion.Max >= 6 &&
+			desc.WireVersion != nil &&
```

Comment:
See my previous comment. I think we can remove the `desc.WireVersion != nil` piece of this conditional.